### PR TITLE
Implementation of the information of characters

### PR DESCRIPTION
**Pull request by ezequielsetzesglobant**
The functionality of displaying the information of the selected character in the RecyclerView in a DialogFragment has been implemented.

Pixel
https://user-images.githubusercontent.com/84338513/132512598-9ff69d26-b63a-48be-8938-3db494ca4102.mp4

Nexus
https://user-images.githubusercontent.com/84338513/132512654-43220fec-3434-4eb7-9d9d-dd5491dcc313.mp4

### CODE REVIEW

**Review by lucianoeli — APPROVED** [review]

**Review by SartiMau — COMMENTED** [review]
please transform the background_item_recycler_view_fragment.jpg to webp

File: app/src/main/java/com/example/cleanarchitecture/fragment/CharacterInformationFragment.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                } else {
+                    textViewFragmentDescription.text = context?.getString(R.string.text_view_default_description_message_fragment)
+                }
+                Glide.with(root)
+                    .load("${it.thumbnail.path}.${it.thumbnail.extension}")
+                    .into(imageViewFragmentIcon)
+                recyclerViewFragmentComics.layoutManager = LinearLayoutManager(context, LinearLayoutManager.HORIZONTAL, false)
+                recyclerViewFragmentComics.adapter = ComicAdapter(it.comics.items)
+            }
+        }
```
Comment (mauroLonghi) [inline app/src/main/java/com/example/cleanarchitecture/fragment/CharacterInformationFragment.kt:55]: you need to do this into the EVENT , not outside

File: app/src/main/res/layout/fragment_character_information.xml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="@drawable/background_fragment_character_information"
+    tools:context=".fragment.CharacterInformationFragment">
+
+    <ImageButton
+        android:id="@+id/image_button_fragment_close"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:src="@android:drawable/btn_dialog"
```
Comment (mauroLonghi) [inline app/src/main/res/layout/fragment_character_information.xml:14]: you should use @drawable/...

Comment (mauroLonghi) [reply app/src/main/res/layout/fragment_character_information.xml:14]: and the name it not  erpresentative

**Comment by ezequielsetzesglobant** [comment]
the weight of the image could not be changed to take from jpg to webp so it was left as jpg

**Review by SartiMau — APPROVED** [review]

File: app/src/main/res/layout/fragment_character_information.xml (hunk 2 above)
Comment (ezequielsetzesglobant) [reply app/src/main/res/layout/fragment_character_information.xml:14]: done

File: app/src/main/java/com/example/cleanarchitecture/fragment/CharacterInformationFragment.kt (hunk 1 above)
Comment (ezequielsetzesglobant) [reply app/src/main/java/com/example/cleanarchitecture/fragment/CharacterInformationFragment.kt:55]: done

File: app/src/main/java/com/example/cleanarchitecture/viewmodel/CharacterInformationViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.example.cleanarchitecture.utils.Event
+import com.example.domain.entity.CharacterData
+
+class CharacterInformationViewModel : ViewModel() {
+
+    private var mutableComicsState: MutableLiveData<Event<FragmentData>> = MutableLiveData()
+    val liveDataState: LiveData<Event<FragmentData>>
+        get() = mutableComicsState
```
Comment (nicolbidone) [inline app/src/main/java/com/example/cleanarchitecture/viewmodel/CharacterInformationViewModel.kt:13]: you're creating a useless livedata (liveDataState) please convert this into a function

File: app/src/main/java/com/example/cleanarchitecture/fragment/CharacterInformationFragment.kt [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+class CharacterInformationFragment : DialogFragment() {
+
+    private lateinit var adapter: ComicAdapter
+    private lateinit var binding: FragmentCharacterInformationBinding
+    private val viewModel by viewModel<CharacterInformationViewModel>()
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        binding = FragmentCharacterInformationBinding.inflate(inflater, container, false)
+        viewModel.liveDataState.observe(this, { updateUI(it) })
```
[the remark this answers is not in the document]
Comment (ezequielsetzesglobant) [reply app/src/main/java/com/example/cleanarchitecture/fragment/CharacterInformationFragment.kt:30]: done

File: app/src/main/java/com/example/cleanarchitecture/viewmodel/CharacterInformationViewModel.kt (hunk 3 above)
Comment (ezequielsetzesglobant) [reply app/src/main/java/com/example/cleanarchitecture/viewmodel/CharacterInformationViewModel.kt:13]: done

**Review by mauroLonghi — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Review by nicolbidone — APPROVED** [review]

**Pull request merged**